### PR TITLE
Superblock

### PR DESCRIPTION
**Pull request by hoene**
supporting the latest netcdf libraries.

### CODE REVIEW

File: src/hdf/superblock.c [hunk 1]
```diff
@@ -4,85 +4,176 @@
...
+  memset(superblock, 0, sizeof(*superblock));
+
+  /* signature */
+  if (fread(buf, 1, 8, reader->fhd) != 8 ||
+      strncmp("\211HDF\r\n\032\n", buf, 8)) {
+    log("file does not have correct signature");
+    return MYSOFA_INVALID_FORMAT;
+  }
+
+  /* read version of superblock, must be 2 */
```
Comment (vsaari) [inline src/hdf/superblock.c]: comment out of date

File: src/hdf/fractalhead.c [hunk 2]
```diff
@@ -4,334 +4,342 @@
...
+    typeandversion = (uint8_t)fgetc(reader->fhd);
+    offset = readValue(reader, offset_size);
+    length = readValue(reader, length_size);
+    if (offset > 0x10000000 || length > 0x10000000)
+      return MYSOFA_UNSUPPORTED_FORMAT;
+
+    log(" %d %4" PRIX64 " %" PRIX64 " %08lX\n", typeandversion, offset, length,
+        ftell(reader->fhd));
+
+    /* TODO: for the following part, the specification is incomplete */
```
Comment (vsaari) [inline src/hdf/fractalhead.c]: Could the unspecified part be specified in the netcdf4 format ? Just a guess, I didn't find anything with quick googling

File: src/hdf/dataobject.c [hunk 3]
```diff
@@ -278,129 +357,192 @@ static int readOHDRHeaderMessageDatatype(struct READER *reader,
...
+
+  version = (uint8_t)fgetc(reader->fhd);
+  switch (version) {
+  case 1:
+  case 2:
+    return readOHDRHeaderMessageDataFill1or2(reader);
+  case 3:
+    return readOHDRHeaderMessageDataFill3(reader);
+  default:
+    mylog("object OHDR data storage fill value message must have version 3 not "
```
Comment (hoene) [inline src/hdf/dataobject.c]: 1,2,3

File: src/hdf/dataobject.c [hunk 4]
```diff
@@ -278,129 +357,192 @@ static int readOHDRHeaderMessageDatatype(struct READER *reader,
...
+}
 
-	version = (uint8_t) fgetc(reader->fhd);
-	if (version != 3) {
-		log(
-				"object OHDR data storage fill value message must have version 3 not %d\n",
-				version);
-		return MYSOFA_INVALID_FORMAT;
-	}
+static int readOHDRHeaderMessageDataFillOld(struct READER *reader) {
```
Comment (hoene) [inline src/hdf/dataobject.c]: still needed?

File: src/hdf/dataobject.c [hunk 5]
```diff
@@ -440,209 +582,213 @@ static int readOHDRHeaderMessageGroupInfo(struct READER *reader,
...
+    /*
+     * 000036e3   67 0e 00 00 00  00 00 00 00 00 00 00 00  |...g............|
+     000036f0  00 00 00
+     */
+  case 6:
+    /* TODO unclear spec */
+    mylog("COMPONENT todo %lX %d\n", ftell(reader->fhd), dt->size);
+    if (fseek(reader->fhd, dt->size, SEEK_CUR))
+      return errno;
+    //  return MYSOFA_INVALID_FORMAT;
```
Comment (hoene) [inline src/hdf/dataobject.c]: remove comment

File: src/hdf/fractalhead.c [hunk 6]
```diff
@@ -4,334 +4,452 @@
...
+    6f 6e 73 00  |....Conventions.| 000027c0  00 00 00 00 13 00 00 00  04 00 00
+    00 01 00 00 00  |................| 000027d0  00 00 00 00 53 4f 46 41  01 00
+    08 00 08 00 08 00  |....SOFA........| 000027e0  56 65 72 73 69 6f 6e 00  13
+    00 00 00 03 00 00 00  |Version.........| 000027f0  01 00 00 00 00 00 00 00
+    31 2e 30 01 00 10 00 08  |........1.0.....| 00002800  00 08 00 53 4f 46 41
+    43  6f 6e 76 65 6e 74 69 6f  |...SOFAConventio| 00002810  6e 73 00 13 00 00
+    00 13  00 00 00 01 00 00 00 00  |ns..............| 00002820  00 00 00 53 69
+    6d 70 6c  65 46 72 65 65 46 69 65  |...SimpleFreeFie|
+    */
+        if (!(name = malloc(100)))
```
Comment (hoene) [inline src/hdf/fractalhead.c]: use constant not 100

File: src/hdf/fractalhead.c [hunk 7]
```diff
@@ -4,334 +4,452 @@
...
+            free(name);
+            return MYSOFA_READ_ERROR;
+          }
+          name[i] = c;
+          if (len < 0 && c == 0)
+            len = i;
+          if (c == 0x13)
+            break;
+        }
+        name = realloc(name, len + 1);
```
Comment (hoene) [inline src/hdf/fractalhead.c]: error check

**Pull request merged**